### ISSUE TITLE
Housekeeping on client connections

### ISSUE THREAD

**Issue by sanity** [kwebio/core#2]
Information on WebSocket connection to clients is kept in KWeb.clients, currently a ConcurrentHashMap.

This means that they won't be deleted until the server is restarted.  We should keep track of when clients disconnect from the server and delete any associated data.

**Comment by sanity** [comment]
Per @swishy, Wasabi doesn't yet have a callback for a websocket connection closing, but he is going to add it.

**Comment by swishy** [comment]
See wasabi #108 , thanks Ian.

**Comment by sanity** [comment]
Note that we also need a mechanism to allow resources to be cleaned up when a client connection is terminated.  This could be a callback specified within the buildPage function passed to the KWeb constructor.  Here is what it might look like:

```kotlin
KWeb() {
   // ...
  onDisconnect() {
    // Cleanup resources here
  }
}
```

We've added a `onDisconnect` method to the RootReceiver.

**Comment by sanity** [comment]
This was fixed in commit bb43a39290b968ea6337eccba615f34a17608cae (sorry about the crappy commit message).
